src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -333,11 +359,13 @@ public class Memory {
     }
 
     private void Write(uint address, byte value) {
+        address &= A20AddressMask;
```

Comment:
Would be nice if performance is not too much impacted by this to also do a modulo memory size (which would differ between a20 enabled or not?) since this is how it works in hardware and some games may use this obscure feature.
If you choose to implement this, it could make sense to move all this to a class that just computes the physical address based on this and A20. Could be useful for rewritten code as well to have such class.

Reply:
Done (putting the A20 Gate into a class) and done (using modulo on top). :)

Reply:
Actually, the modulo is an effect of the AND performed by the hardware, which comes down to (address &= ((2**addressBits) -1))

Example: 
Using 20 bits (A20 disabled):
2**20  = 0x100000
0x100000 - 1 = 0xFFFFF
0x100123 & 0xFFFFF = 0x123

Using 21 bits (A20 enabled):
2**21 = 0x200000
0x200000 - 1 = 0x1FFFFF
0x100123 & 0x1FFFFF = 0x100123

Reply:
Done (calculations simplified).